subsys/nfc/lib/Kconfig

```diff
@@ -21,6 +21,130 @@ config NFCT_IRQ_PRIORITY
 	  Sets NFC interrupt priority.
 	  Levels are from 0 (highest priority) to 6 (lowest priority)
 
+config NFC_THREAD_CALLBACK
+	bool "Thread context for NFC callbacks"
+	select RING_BUFFER
+	default y
+	help
+	  Use this option to decouple the user NFC callback
+	  from the NFC interrupt context. When enabled, the user
+	  callback is called from a thread (System Workqueue
+	  or a dedicated thread depending on the CONFIG_NFC_OWN_THREAD
+	  setting) instead of being called from the interrupt context.
+	  Using this option ensures robust NFC communication
+	  (deterministic time of the interrupt handling) and allows
+	  deferred callback execution, while it results in higher memory
+	  footprint and CPU load.
+
+if NFC_THREAD_CALLBACK
+
+config HEAP_MEM_POOL_SIZE
+	default 512
+
+config NFC_RING_SIZE
+	int "Size of ring buffer in bytes."
+	default 2048
+	help
+	  Size of the ring buffer.
+	  The specified size is expressed as n bytes.
+
+config NFC_LIB_CTX_MAX_SIZE
+	int "Maximum size of NFC library context in bytes."
+	default 16
+	help
+	  Maximum size of the NFC library context in bytes.
+	  The specified size should be divisible by 4.
+
+config NFC_OWN_THREAD
+	bool "Call NFC callback from its own thread"
+	default n
+	help
+	  Use dedicated a thread to call NFC callback.
+	  Otherwise use the system workqueue thread.
+
+if NFC_OWN_THREAD
+
+config NFC_THREAD_STACK_SIZE
+	int "Size of NFC thread stack"
+	default 256
+	help
+	  Size of the NFC dedicated thread stack in bytes.
+
+config NFC_THREAD_PRIORITY
+	int "Priority of NFC callback thread"
+	default -1
```

Comment:
Must it be a cooperative thread? If so, it may be worth adding a note about it.

Reply:
It does not have to be a cooperative thread, but processing the nfc callbacks without cooperation with other threads may decrease latency and fifo usage.